src/methods/delta/signDeltaOrder.ts

```diff
@@ -0,0 +1,54 @@
+import { deriveCompactSignature } from '../../helpers/misc';
+import type { ConstructProviderFetchInput } from '../../types';
+import { SignableDeltaOrderData } from './helpers/buildDeltaOrderData';
+import { sanitizeDeltaOrderData } from './helpers/misc';
+
+export type SignLimitOrderFunctions = {
+  signLimitOrder: (
+    signableOrderData: SignableDeltaOrderData
+  ) => Promise<string>;
+};
+
+type SignDeltaOrder = (
+  signableOrderData: SignableDeltaOrderData
+) => Promise<string>;
+
+export type SignDeltaOrderFunctions = {
+  signDeltaOrder: SignDeltaOrder;
+};
+
+// returns whatever `contractCaller` returns
+// to allow for better versatility
+export const constructSignDeltaOrder = (
+  options: Pick<
+    ConstructProviderFetchInput<any, 'signTypedDataCall'>,
+    'contractCaller'
+  >
+): SignDeltaOrderFunctions => {
+  const signDeltaOrder: SignDeltaOrder = async (typedData) => {
+    // types allow to pass OrderData & extra_stuff, but tx will break like that
+    const typedDataOnly: SignableDeltaOrderData = {
+      ...typedData,
+      data: sanitizeDeltaOrderData(typedData.data),
+    };
+    const signature = await options.contractCaller.signTypedDataCall(
+      typedDataOnly
+    );
+
+    if (signature.length > 132) {
```

Comment:
In case of safe multisig - signature is of 132 length, so most likely(I am not sure) you need to check for `>=`

Reply:
132 for one signer only, the more signers the longer the signature